### PR TITLE
 Ensure that nil parameters don't defeat locale handling

### LINKED ISSUE #109: Passing nil request parameters defeats appendsPreferredLanguageLocale
### Expected behavior

By default, `WordPressComRestApi` automatically appends `locale` to requests when `appendsPreferredLanguageLocale` is enabled (i.e., the default state). For requests where `nil` parameters are passed to `WordPressComRestApi.buildRequestURLFor(path:parameters:)`, locale is does not honor that request.

Requests that specify `locale` are not impacted.

### Actual behavior

Passing `nil` to `WordPressComRestApi.buildRequestURLFor(path:parameters:)` should result in a URL that includes locale when `appendsPreferredLanguageLocale` is enabled.

##### WordPressKit Environment
 - WordPressKit Version : `WordPressKit` v2.0.0 & later
 - OS : iOS 12.x
 - Device : all devices
 - Xcode Version : Xcode 10.2 beta

##### Additional Context
See related discussion in `p1551132833014600-slack-build-and-ship`.

### PR DESCRIPTION
**Pull request by stevebaranski**
### Description

Fixes #109, where it was observed that `nil` parameters could prevent the automatic insertion of a locale parameter.

### Testing Details

- Checkout the branch and confirm that tests pass.
- Smoke test the changes with `issue/wpkit-109-validation`

- [X] Please check here if your pull request includes additional test coverage.

### CODE REVIEW

**Comment by stevebaranski** [comment]
@aerych would you have a moment to take a look at this PR?

**Comment by jklausa** [comment]
afraid I've stolen the -beta.2 number off of you already ;P

File: WordPressKit.podspec [hunk 1]
```diff
@@ -1,6 +1,6 @@
 Pod::Spec.new do |s|
   s.name          = "WordPressKit"
-  s.version       = "2.1.1-beta.2"
+  s.version       = "2.1.1-beta.3"
```
Comment (aerych) [inline WordPressKit.podspec]: I'm not sure how we should best handle this.  We're wanting to incorporate this change into WPiOS 11.9, but it looks like its branched from `develop` which incorporates other changes as well.  I think we'd want to branch off of either `master` (currently at 2.1.0), or the 2.1.0 tag instead. 

The question then is what the version would be, `2.1.0.1 maybe?  Pinging @loremattei and @jkmassel who are wrangling the 11.9 code freeze and release respectively for guidance.

Comment (jkmassel) [reply WordPressKit.podspec]: I think `2.1.0.1` is the way to go, yes

Comment (stevebaranski) [reply WordPressKit.podspec]: Fair enough. I will close & resubmit.

**Pull request closed without merging**